src/BSON/Binary.c

```diff
@@ -46,10 +46,12 @@ PHONGO_API zend_class_entry *php_phongo_binary_ce;
 
 zend_object_handlers php_phongo_handler_binary;
 
-/* Initialize the object from a string and return whether it was successful. */
-static bool php_phongo_binary_init(php_phongo_binary_t *intern, const char *data, phongo_zpp_char_len data_len, phongo_long type)
+/* Initialize the object and return whether it was successful. An exception will
+ * be thrown on error. */
+static bool php_phongo_binary_init(php_phongo_binary_t *intern, const char *data, phongo_zpp_char_len data_len, phongo_long type TSRMLS_DC)
 {
 	if (type < 0 || type > UINT8_MAX) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected type to be an unsigned 8-bit integer, %" PHONGO_LONG_FORMAT " given", type);
```

Comment:
This message was previously only used in the constructor; however, the exception may now be thrown from `__set_state()` or `unserialize()`. What do you think about putting the class name in the message?

I've started doing this with other new messages (e.g. Javascript now allowing null bytes in its code argument) for added clarity, but did not want to change existing messages before discussing.